### PR TITLE
How to Find the First Match Materials

### PR DESCRIPTION
**Pull request by iansedano**
**Where to put new files:**

- New files should go into a top-level subfolder, named after the article slug. For example: `my-awesome-article`

**How to merge your changes:** 

1. [Make sure the CI code style tests all pass (+ run the automatic code formatter if necessary).](https://github.com/realpython/materials/blob/master/README.md)
2. Find an RP Team member on Slack and ask them to review & approve your PR.
3. Once the PR has one positive ("approved") review, GitHub lets you merge the PR.
4. 🎉

### CODE REVIEW

File: python-first/get_first.py [hunk 1]
```diff
@@ -4,20 +4,22 @@
...
+            gen = (elem for elem in iterable if key(elem))
         case (False, True):
-            gen = (val for val in iterable if key(val) == value)
+            gen = (elem for elem in iterable if key(elem) == value)
         case (True, False):
-            gen = (val for val in iterable if val)
+            gen = (elem for elem in iterable if elem)
         case (False, False):
-            gen = (val for val in iterable if val == value)
+            gen = (elem for elem in iterable if elem == value)
```
Comment (gahjelle) [inline python-first/get_first.py:13]: I think I find `elem` or even `element` a better name for each element in iterable than juggling `val` and `value`. What do you think?

File: python-first/get_first.py [hunk 2]
```diff
@@ -4,20 +4,22 @@
...
 
     return next(gen, default)
 
 
 if __name__ == "__main__":
-    target_match = {"country": "Norway", "population": 5311916}
-    print(get_first(countries, target_match))
+    print(get_first(countries))
```
Comment (gahjelle) [inline python-first/get_first.py:19]: I added the example of just getting the first _truthy_ country as well.

File: python-first/get_first_alt.py [hunk 3]
```diff
@@ -1,20 +1,22 @@
...
     else:
-        gen = (val for val in iterable if val)
+        gen = (elem for elem in iterable if elem)
 
     return next(gen, default)
 
 
 if __name__ == "__main__":
-    target_match = {"country": "Norway", "population": 5311916}
-    print(find_match_gen(countries, target_match))
```
Comment (gahjelle) [inline python-first/get_first_alt.py]: This didn't work as intended. It returned `Austria` and not Norway.

File: python-first/get_first_loop.py [hunk 4]
```diff
@@ -2,24 +2,26 @@
...
 
-        if (value is None and _val) or (value is not None and _val == value):
-            return val
+        if (value is None and _elem) or (value is not None and _elem == value):
+            return elem
     return default
 
 
 if __name__ == "__main__":
-    target_match = {"country": "Puerto Rico", "population": 3195153}
```
Comment (gahjelle) [inline python-first/get_first_loop.py]: Puerto Rico is no longer part of countries

**Review by KateFinegan — APPROVED** [review]

**Pull request merged**